tools/config/__init__.py

```diff
@@ -632,10 +631,32 @@ def get_all_active_memories(self, memory_list):
         active_memory_counter = 0
         # Find which memory we are dealing with, RAM/ROM
         active_memory = 'RAM' if any('RAM' in mem_list for mem_list in memory_list) else 'ROM'
-        cmsis_part = self._get_cmsis_part()
+        
+        try:
+            cmsis_part = self._get_cmsis_part()
+        except ConfigException:
+            """ If the target doesn't exits in cmsis, but present in targets.json
+            with ram and rom start/size defined"""
+            if hasattr(self.target, "mbed_ram_start") and \
+               hasattr(self.target, "mbed_rom_start"):
+               mem_start = getattr(self.target, "mbed_" + active_memory.lower() + "_start")
+               mem_size = getattr(self.target, "mbed_" + active_memory.lower() + "_size")
+               available_memories[active_memory] = [mem_start, mem_size]
+               return available_memories
+            else:
+                raise ConfigException("Bootloader not supported on this target. "
+                                      "ram/rom start/size not found in "
+                                      "targets.json.")
+
         present_memories = set(cmsis_part['memory'].keys())
         valid_memories = set(memory_list).intersection(present_memories)
 
+        """For targets that do not have ram/rom defined, create dict with
+        start/size as 0 """
+        if not valid_memories:
+            available_memories[active_memory] = [0, 0]
```

Comment:
Is this how we currently handle this case? Or would raising a `ConfigException` be more appropriate here?

Reply:
This is not how we currently handle this case. Please raise the ConfigException like we had done before.

Reply:
But thats not how we were handling it before. I think my comment is misleading here. This part of code is to handle different memories that exists under RAM/ROM, for example, for STM32L443RC, we do have IROM defined, but we do not have any ram defined. In this case, we still need to create defines, MBED_RAM_START/SIZE, unless we intend to ifdef that in the C code, which is fine by me.

Reply:
Hi

> for STM32L443RC, we do have IROM defined

Issue is know since #5968 
@theotherjimmy